File: frontend/src/app/post/dto/post-preview.dto.ts
export interface PostPreviewDTO {
  id: string;
  title: string;
  content: string;
  type: PostType;
  file?: string;
  upvotes: number;
  downvotes: number;
  username: string;
  subredditName: string;
  createdAt: string;
}

export enum PostType {
  TEXT = 'TEXT',
  LINK = 'LINK',
  IMAGE = 'IMAGE',
  VIDEO = 'VIDEO'
}

File: backend/app/src/main/java/com/redditclone/app/post/application/dto/PostPreviewDTO.java
package com.redditclone.app.post.application.dto;

import com.redditclone.app.post.domain.PostType;
import lombok.Data;

import java.time.Instant;
import java.util.UUID;

@Data
public class PostPreviewDTO {
    private UUID id;
    private String title;
    private String content;
    private PostType type;
    private byte[] file;
    private int upvotes;
    private int downvotes;
    private String username;
    private String subredditName;
    private Instant createdAt;
}
